### PR TITLE
Pal 956 ams disabled test

### PR DESCRIPTION
**Pull request by developer01189998819991197253**

### CODE REVIEW

File: attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/service/AttributeMaskingServiceErrorTest.java [hunk 1]
```diff
@@ -111,11 +112,16 @@ void testMaskingFailure() {
...
+                .isNotNull()
+                .extracting(AuditErrorMessage::getError)
+                .extracting(Throwable::getMessage)
                 .isEqualTo("Cannot mask");
 
-        assertThat(subject.getAttributeMaskingResponse())
+        assertThat(subject)
+                .extracting(AuditableAttributeMaskingResponse::getAttributeMaskingResponse)
                 .as("verify that auditable object has no payload")
                 .isNull();
```
Comment (pd104923) [inline attribute-masking-service/src/component-tests/java/uk/gov/gchq/palisade/component/attributemask/service/AttributeMaskingServiceErrorTest.java]: Might be worth moving the `.as()` so it's in the same place in the tests. More of an aesthetic change though.
```suggestion
        assertThat(subject)
                .as("verify that auditable object has no payload")
                .extracting(AuditableAttributeMaskingResponse::getAttributeMaskingResponse)
                .isNull();
```

File: attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/ContractTestData.java [hunk 2]
```diff
@@ -80,7 +88,7 @@ public T apply(final T record, final User user, final Context context) {
...
+    public static final Function<Integer, String> REQUEST_FACTORY_JSON = i -> String.format("{\"userId\":\"test-user-id\",\"resourceId\":\"/test/resourceId\",\"context\":{\"class\":\"uk.gov.gchq.palisade.Context\",\"contents\":{\"purpose\":\"test-purpose\"}},\"user\":{\"userId\":{\"id\":\"test-user-id\"},\"roles\":[],\"auths\":[],\"class\":\"uk.gov.gchq.palisade.User\"},\"resource\":{\"class\":\"uk.gov.gchq.palisade.resource.impl.FileResource\",\"id\":\"/test/resourceId\",\"attributes\":{},\"connectionDetail\":{\"class\":\"uk.gov.gchq.palisade.service.SimpleConnectionDetail\",\"serviceName\":\"test-data-service\"},\"parent\":{\"class\":\"uk.gov.gchq.palisade.resource.impl.SystemResource\",\"id\":\"/test/\"},\"serialisedFormat\":\"avro\",\"type\":\"%d\"},\"rules\":{\"message\":\"no rules set\",\"rules\":{\"test-rule\":{\"class\":\"uk.gov.gchq.palisade.contract.attributemask.ContractTestData$PassThroughRule\"}}}}", i, i);
```
Comment (pd104923) [inline attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/ContractTestData.java]: One of the `i`s can be removed from the end of this line. There only needs to be 1 argument.

File: attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+                                                .as("Assert that after extracting the AuditErrorMessage, it is as expected")
+                                                .extracting(ConsumerRecord::value)
+                                                .usingRecursiveComparison()
+                                                .ignoringFieldsOfTypes(Throwable.class)
+                                                .ignoringFields("timestamp")
+                                                .isEqualTo(ContractTestData.AUDIT_ERROR_MESSAGE),
+
+                                        () -> assertThat(error.value().getError())
+                                                .as("Assert that the error message inside the AuditErrorMessage is the same")
+                                                .isExactlyInstanceOf(Throwable.class)
```
Comment (pd104923) [inline attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java:156]: Is there any way of checking the exact error class?

**Review by pd104923 — CHANGES_REQUESTED** [review]
A couple of little questions. 
I'm not sure if the KafkaHealthIdicator needed to be changed as it has a consumer group that can be checked rather than checking the topics.

File: attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java (hunk 3 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java:156]: The AMS doesnt throw a unique error, unlike the other services there isnt a FailedToPersistError.class, it just throws a generic exception which calls its super() until it finally produces a Throwable.class erorr message. In other cases you could check the stack trace for the original error message but this doesnt apply here

File: attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/ContractTestData.java (hunk 2 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/ContractTestData.java]: removed the extra `i`

File: attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java (hunk 3 above)
Comment (pd104923) [reply attribute-masking-service/src/contract-tests/java/uk/gov/gchq/palisade/contract/attributemask/kafka/KafkaErrorContractTest.java:156]: Fair enough

File: attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java [hunk 4]
```diff
@@ -35,22 +35,20 @@
 import java.util.concurrent.TimeoutException;
 
 /**
- * Kafka health indicator. Check that the consumer group can be accessed and is registered with the cluster,
+ * Kafka health indicator. Check that the producer group can be accessed and is registered with the cluster,
```
Comment (pd104923) [inline attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java]: This change should be reverted as well.

**Review by pd104923 — CHANGES_REQUESTED** [review]
One minor change

File: attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java (hunk 4 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java]: reverted

**Review by pd104923 — DISMISSED** [review]
Looking good now

**Review by dbz120896 — DISMISSED** [review]

File: attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java [hunk 5]
```diff
@@ -80,23 +79,27 @@ public Health health() {
...
 
             LOGGER.debug("Kafka consumer group ({}) state: {}", groupId, consumerGroupDescription.state());
 
+
             if (consumerGroupDescription.state() == ConsumerGroupState.STABLE) {
-                return consumerGroupDescription.members().stream()
+                boolean assignedGroupPartition = consumerGroupDescription.members().stream()
                         .noneMatch(member -> (member.assignment() == null || member.assignment().topicPartitions().isEmpty()));
+                if (!assignedGroupPartition) {
+                    LOGGER.error("Failed to find kafka topic-partition assignments");
```
Comment (dev930018) [inline attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java]:
```suggestion
                    LOGGER.error("Failed to find kafka group id assignments");
```

File: attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java [hunk 6]
```diff
@@ -80,23 +79,27 @@ public Health health() {
...
+                }
+                return assignedGroupPartition;
             }
         } catch (InterruptedException | ExecutionException | TimeoutException e) {
```
Comment (dev930018) [inline attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java]: The `InterruptedException` should result in a `Thread.currentThread.interrupt()`, but not an `ExecutionException` or `TimeoutException`

**Review by dev930018 — CHANGES_REQUESTED** [review]

File: attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java (hunk 6 above)
Comment (developer01189998819991197253) [reply attribute-masking-service/src/main/java/uk/gov/gchq/palisade/service/attributemask/service/KafkaHealthIndicator.java]: updated

**Review by dev930018 — APPROVED** [review]

**Pull request merged**